apps/aecore/src/aec_headers.erl

```diff
@@ -272,10 +275,37 @@ validate_pow({#header{nonce        = Nonce,
             {error, incorrect_pow}
     end.
 
--spec validate_time({header(), aec_governance:protocols()}) ->
-                           ok | {error, block_from_the_future}.
-validate_time({#header{time = Time}, _}) ->
-    MaxAcceptedTime = aeu_time:now_in_msecs() + ?ACCEPTED_FUTURE_KEY_BLOCK_TIME_SHIFT,
+-spec validate_median_time({header(), aec_governance:protocols()}) ->
+        ok | {error, block_from_the_past}.
+validate_median_time({Header = #header{time = Time}, _}) ->
+    case aec_chain_state:median_timestamp(Header) of
+        {ok, MTime} when Time > MTime -> ok;
+        {ok, _MTime}                  -> {error, block_from_the_past};
+        error                         -> ok %% We can't know yet - checked later
```

Comment:
I see that "later" is done by assert_key_block_time alongside other checks when block connected to genesis. Ok.